### ISSUE TITLE
[UI] [Linux] - Issues with `Emulation Settings` windows.

### ISSUE THREAD

**Issue by aphirst** [PCSX2/pcsx2#219]
**System**:
* Arch Linux x86_64
* Thinkpad X220, with an Intel HD 3000
* pcsx2-git d247a79e489bb27b68a76741b7a32c0b61a2839e
* no DE, just openbox

I'm having two minor issues with the UI, which are not present in the 1.2.2 stable build in the repository.

1. After opening the "Emulation Settings" window for a second time (by going `Config -> Emulation Settings`), the UI elements at the bottom compress vertically so as to be almost unusable.

2. The ISO File Selector dialog does not correctly filter filetypes, i.e. no files are shown in a folder which in fact does contain compatible ISO files.

Screenshots demonstrating these two problems can be seen at http://postimg.org/gallery/22yg4684g/

If there is any other information that would be needed to further diagnose, please do not hesitate to ask.

**Comment by gregory38** [comment]
I made some fix for issue 2 so it might work better. Could you test latest git.

**Comment by aphirst** [comment]
@gregory38 I can confirm that the ISO file selector now seems to work correctly in pcsx2-git.

The issue with the UI elements (buttons) becoming compressed on subsequent openings of dialogs is still present.

**Comment by gregory38** [comment]
I don't have any issue here. Do you use wx2.8 or wx3.0? You can see it with ldd command on pcsx2 binary.

**Comment by aphirst** [comment]
@gregory38 
```
[adam@iwakura ~]$ ldd /usr/bin/pcsx2 | grep wx
	libwx_baseu-3.0.so.0 => /usr/lib32/libwx_baseu-3.0.so.0 (0xf7457000)
	libwx_gtk2u_core-3.0.so.0 => /usr/lib32/libwx_gtk2u_core-3.0.so.0 (0xf6e60000)
	libwx_gtk2u_adv-3.0.so.0 => /usr/lib32/libwx_gtk2u_adv-3.0.so.0 (0xf6c9d000)
```
I believe `3.0`.

**Comment by gregory38** [comment]
Try a build with wx 2.8

**Comment by aphirst** [comment]
I'll try that as soon as I work out how to modify [the AUR package](https://aur.archlinux.org/packages/pcsx2-git/) to pull `2.8` and set whatever build flags are necessary.

**Comment by gregory38** [comment]
PCSX2 moved to 3.0 :p But it might help to understand the source of the problem. Otherwise try a debug build, maybe a wxWidget assertion will be hit.

**Comment by gregory38** [comment]
To keep a trace of current trials

```
diff --git a/common/src/Utilities/wxHelpers.cpp b/common/src/Utilities/wxHelpers.cpp
index a0d189e..e58f38b 100644
--- a/common/src/Utilities/wxHelpers.cpp
+++ b/common/src/Utilities/wxHelpers.cpp
@@ -330,6 +330,7 @@ void wxDialogWithHelpers::OnOkCancel( wxCommandEvent& evt )
 void wxDialogWithHelpers::AddOkCancel( wxSizer &sizer, bool hasApply )
 {
        wxStdDialogButtonSizer& s_buttons( *new wxStdDialogButtonSizer() );
+       s_buttons.SetMinSize(wxSize(50,50));
 
        s_buttons.AddButton( new wxButton( this, wxID_OK ) );
        s_buttons.AddButton( new wxButton( this, wxID_CANCEL ) );
@@ -354,6 +355,8 @@ void wxDialogWithHelpers::AddOkCancel( wxSizer &sizer, bool hasApply )
        // create a sizer to hold the help and ok/cancel buttons, for platforms
        // that need a custom help icon.  [fixme: help icon prolly better off somewhere else]
        wxFlexGridSizer& flex( *new wxFlexGridSizer( 2 ) );
+       flex.SetFlexibleDirection(wxHORIZONTAL);
+       flex.SetMinSize(wxSize(0, 100));
        flex.AddGrowableCol( 0, 1 );
        flex.AddGrowableCol( 1, 15 );
 

```

**Comment by gregory38** [comment]
So the issue appears on my computer on a different box. For the history
```
Delete bin/logs directory.
Create a broken link from bin/logs to a non-existing directory
Run PCSX2, it will pop-up a message saying that it can't create the log. Ok/Cancel button will be flat.
```

@turtleli 
You did a wonderful jobs with others WX issues. Feel free to fix it if you can reproduce it.

**Comment by turtleli** [comment]
I looked at it before but I didn't figure anything out. The bug affects me on my Fedora 22 VM but I don't think it affects my Slackware boot.

I did notice this time that when the bug happens, the dialog boxes are smaller by 1 button height. Maybe that'll help anyone who is also looking at this issue.

**Comment by turtleli** [comment]
Well, I've been investigating more, and I think I can fix this.

I still can't figure out how some of the custom GUI stuff works, but I can solve the problem using the standard wxWidgets API stuff, so I'll probably do that. It'll probably take a while though since I have to change things one dialog at a time.

**Comment by turtleli** [comment]
@aphirst 
I've committed a fix for this, can you test? It seems to work on my Fedora VM, the testcase gregory had above also works on my Slackware boot (though there are other problems with the portable mode settings dialog).

**Comment by aphirst** [comment]
@turtleli I had a play about with it and it seems to work! I can't guarantee I checked everything, naturally, but I didn't see any weirdness.

**Comment by turtleli** [comment]
That's good. :)

I'll leave the issue open for another week or two just in case anyone finds I missed something.

**Comment by turtleli** [comment]
Fixed by commit 31a505768de78af7b5dcd54620a0f10cea250b3f
